### PR TITLE
PIN-3839: updated calls to authorizationProcess

### PR DESCRIPTION
**Pull request by AsterITA**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala [hunk 1]
```diff
@@ -6,34 +6,37 @@ import it.pagopa.interop.commons.utils.TypeConversions._
...
+  UserRegistryService
 }
 import it.pagopa.interop.backendforfrontend.api.ClientsApiService
 import akka.http.scaladsl.marshalling.ToEntityMarshaller
 import akka.http.scaladsl.server.Directives.onComplete
 import akka.http.scaladsl.server.Route
 import it.pagopa.interop.backendforfrontend.error.Handlers.handleError
 import it.pagopa.interop.commons.logging.{CanLogContextFields, ContextFieldsToLog}
-import it.pagopa.interop.backendforfrontend.model._
+import it.pagopa.interop.backendforfrontend.model.{RelationshipState, _}
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala]:
```suggestion
import it.pagopa.interop.backendforfrontend.model._
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala [hunk 2]
```diff
@@ -282,7 +286,12 @@ final case class ClientsApiServiceImpl(
     val result: Future[PublicKeys] = for {
       clientUuid     <- clientId.toFutureUUID
       readClientKeys <- authorizationProcessService.getClientKeys(clientUuid, Seq.empty)
-      keys           <- Future.traverse(readClientKeys.keys)(decorateKey)
+      keys           <- Future.traverse(readClientKeys.keys)(key =>
+        for {
+          user         <- userRegistryService.findById(key.relationshipId)
+          decoratedKey <- decorateKey(key, user)
+        } yield decoratedKey
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala]: I suggest a separated function

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala [hunk 3]
```diff
@@ -365,7 +378,12 @@ final case class ClientsApiServiceImpl(
       clientUuid       <- clientId.toFutureUUID
       relationshipUuid <- relationshipId.toFutureUUID
       readClientKeys   <- authorizationProcessService.getClientKeys(clientUuid, Seq(relationshipUuid))
-      keys             <- Future.traverse(readClientKeys.keys)(decorateKey)
+      keys             <- Future.traverse(readClientKeys.keys)(key =>
+        for {
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala]: same

File: src/main/scala/it/pagopa/interop/backendforfrontend/service/types/AuthorizationProcessServiceTypes.scala [hunk 4]
```diff
@@ -96,18 +94,21 @@ object AuthorizationProcessServiceTypes {
       CompactClient(id = c.client.id, name = c.client.name, hasKeys = c.keys.nonEmpty)
   }
 
-  implicit class OperatorDetailsConverter(private val od: AuthorizationProcess.OperatorDetails) extends AnyVal {
-    def toApi: SelfcareUser =
-      SelfcareUser(relationshipId = od.relationshipId, familyName = od.familyName, name = od.name)
+  implicit class OperatorDetailsConverter(private val u: UserResource) extends AnyVal {
+    def toApi(relationshipId: UUID): SelfcareUser = {
+      val name       = u.name.map(_.value).getOrElse("")
+      val familyName = u.familyName.map(_.value).getOrElse("")
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/service/types/AuthorizationProcessServiceTypes.scala]: the previous behaviour was to fail if one of these was missing.
I prefer your softer approach in this case, but (if you haven't yet) may I suggest to ask Ruggero if he's ok with empty values or if he prefers any other placeholder?

Comment (AsterITA) [reply src/main/scala/it/pagopa/interop/backendforfrontend/service/types/AuthorizationProcessServiceTypes.scala]: > the previous behaviour was to fail if one of these was missing. I prefer your softer approach in this case, but (if you haven't yet) may I suggest to ask Ruggero if he's ok with empty values or if he prefers any other placeholder?

His opinion is, if it's an improbable situation, we can set as placeholders `"Utente"` as the `name` and the `User Id` as the `surname` (If both are missing), in order to always distinguish the keys even when we have multiple users without those

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**